fastlane/lib/fastlane/helper/sh_helper.rb

```diff
@@ -25,20 +27,20 @@ def self.sh_control_output(command, print_command: true, print_command_output: t
       Encoding.default_external = Encoding::UTF_8
       Encoding.default_internal = Encoding::UTF_8
 
-      command = command.join(' ') if command.kind_of?(Array) # since it's an array of one element when running from the Fastfile
+      command = command.map(&:to_s).map(&:shellescape).join(' ') if command.kind_of?(Array) # since it's an array of one element when running from the Fastfile
       UI.command(command) if print_command
 
       result = ''
       if Helper.sh_enabled?
         exit_status = nil
-        IO.popen(command, err: [:child, :out]) do |io|
+        Open3.popen2e(command) do |stdin, io, thread|
           io.sync = true
           io.each do |line|
             UI.command_output(line.strip) if print_command_output
             result << line
           end
           io.close
```

Comment:
Do you have to close stdout explicitly here? The docs for `popen2e` seem to suggest that closing is not necessary in the block form